tests/e2e/prevent_duplicate_cluster_ids.go

```diff
@@ -0,0 +1,207 @@
+		clusterId = e2eVSphere.Config.Global.ClusterID
+		framework.Logf("clusterId: %v", clusterId)
+		revertToOriginalVsphereSecret = false
+		framework.Logf("revertSecret: %v", revertToOriginalVsphereSecret)
+
+		csiNamespace = GetAndExpectStringEnvVar(envCSINamespace)
+		csiDeployment, err := client.AppsV1().Deployments(csiNamespace).Get(
+			ctx, vSphereCSIControllerPodNamePrefix, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		csiReplicas = *csiDeployment.Spec.Replicas
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		framework.Logf("revertSecret: %v", revertToOriginalVsphereSecret)
+		if !revertToOriginalVsphereSecret {
+			ginkgo.By("Delete vsphere-csi-cluster-id configmap if it exists")
+			_, err := client.CoreV1().ConfigMaps(csiNamespace).Get(ctx,
+				vsphereClusterIdConfigMapName, metav1.GetOptions{})
+			if !apierrors.IsNotFound(err) {
+				err = client.CoreV1().ConfigMaps(csiNamespace).Delete(ctx,
+					vsphereClusterIdConfigMapName, metav1.DeleteOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+			ginkgo.By("Reverting back to original vsphere secret")
+			framework.Logf("clusterId: %v", clusterId)
+			createCsiVsphereSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace, vCenterIP,
+				clusterId, vCenterPort, dataCenter, "")
+
+			ginkgo.By("Restart CSI driver")
+			restartSuccess, err := restartCSIDriver(ctx, client, namespace, csiReplicas)
+			gomega.Expect(restartSuccess).To(gomega.BeTrue(), "csi driver restart not successful")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+	})
+
+	/*
+		Generate unique cluster id through configmap and create workloads
+		1. Create vsphere config secret with no cluster id field.
+		2. Validate that "vsphere-csi-cluster-id" configmap is generated with a unique cluster id.
+		3. Create statefulset with replica 3 and a deployment.
+		4. Verify all PVCs are in bound state and pods are in running state.
+		5. Scale sts replica to 5.
+		6. Verify cns metadata and check if cluster id is populated in cns metadata.
+		7. Clean up the sts, deployment, pods and PVCs.
+
+	*/
+	ginkgo.It("[csi-block-vanilla][csi-file-vanilla] Generate unique cluster id through configmap"+
+		" and create workloads", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// decide which test setup is available to run
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+
+		ginkgo.By("Creating csi config secret with no cluster id field set")
+		createCsiVsphereSecret(client, ctx, vCenterUIUser, vCenterUIPassword, csiNamespace, vCenterIP,
+			"", vCenterPort, dataCenter, "")
+
+		ginkgo.By("Restart CSI driver")
+		restartSuccess, err := restartCSIDriver(ctx, client, namespace, csiReplicas)
+		gomega.Expect(restartSuccess).To(gomega.BeTrue(), "csi driver restart not successful")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
put this in a separate call, I see this in many different places

Reply:
Done